### PR TITLE
Fixes #21175: Impossible to put a Rule in a subcategory

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/21175

### CODE REVIEW

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

File: webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/RuleTests.scala [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        , RuleCategoryId("cat1")
+        , RuleCategoryId("cat2")
+        , RuleCategoryId("cat3")
+        , RuleCategoryId("cat4")
+        , RuleCategoryId("subcat4")
+        , RuleCategoryId("subsubcat4")
+      )
+      restTestSetUp.ruleApiService14.listCategoriesId(root) shouldEqual(listCatIds)
+    }
+
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/RuleTests.scala:106]: you should add a test that you only get cat4 and children if starting from cat4 (locality) and that you correctly get nothing if non existing cat and that you only get one cat starting from cat1

**Review by fanf — COMMENTED** [review]
LGTM modulo missing tests, you can merge after adding them

**Pull request merged**